store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -73,38 +81,46 @@ public String sanitize() throws UnsupportedEncodingException {
         final StringBuilder htmlBuilder = new StringBuilder(html.length());
         // create the renderer that will write the sanitized HTML to the builder
         final HtmlStreamRenderer renderer = HtmlStreamRenderer.create(htmlBuilder,
-            Handler.PROPAGATE,
-            // log errors resulting from exceptionally bizarre inputs
-            new Handler<String>() {
+                Handler.PROPAGATE,
+                // log errors resulting from exceptionally bizarre inputs
+                new Handler<String>() {
 
-                public void handle(final String x) {
-                    throw new AssertionError(x);
-                }
-            });
+            public void handle(final String x) {
+                throw new AssertionError(x);
+            }
+        });
         // create a thread-specific policy
         instantiatePolicy();
         final Policy policy = POLICY_DEFINITION.apply(new StyleTagReceiver(renderer));
         // run the html through the sanitizer
-        HtmlSanitizer.sanitize(cleanMalformedHtml(html), policy);
+        runSanitizer(html, policy, cleanData);
         // return the resulting HTML from the builder
         OwaspHtmlSanitizer.zThreadLocal.remove();
-        return htmlBuilder.toString();
+        return htmlBuilder;
+    }
+
+    private void runSanitizer(String str, Policy policy, boolean cleanData) throws UnsupportedEncodingException {
+        if (cleanData) {
+            HtmlSanitizer.sanitize(cleanMalformedHtml(str, false), policy);
+        } else {
+            HtmlSanitizer.sanitize(str, policy);
+        }
     }
 
-    private String cleanMalformedHtml(String str) throws UnsupportedEncodingException {
+    public String cleanMalformedHtml(String str, boolean printBodyOnly) throws UnsupportedEncodingException {
```

Comment:
Does this need to be made public, since it's conditionally being called by the **cleanData** param in the sanitize() method?

Reply:
It is made public as it's being called from `OwaspHtmlSanitizerTest.testBugTSS18004` for verifying extra double quotes are removed.